packages/cli-kit/src/private/node/ui/components/Tasks.tsx

```diff
@@ -0,0 +1,57 @@
+import {TextAnimation} from './TextAnimation.js'
+import useLayout from '../hooks/use-layout.js'
+import useAsync from '../hooks/use-async.js'
+import {Box, Text} from 'ink'
+import React, {useState} from 'react'
+
+const loadingBarChar = '█'
+
+export interface Task {
+  title: string
+  task: () => Promise<void>
+}
+
+export interface Props {
+  tasks: Task[]
+}
+
+const Tasks: React.FC<Props> = ({tasks}) => {
+  const {width} = useLayout()
+  const loadingBar = new Array(width).fill(loadingBarChar).join('')
+  const [currentTask, setCurrentTask] = useState<Task>(tasks[0]!)
+  const [state, setState] = useState<'success' | 'failure' | 'loading'>('loading')
+
+  const runTasks = async () => {
+    for (const task of tasks) {
+      setCurrentTask(task)
+      // eslint-disable-next-line no-await-in-loop
+      await task.task()
```

Comment:
Are there plans to implement a version of this component with tasks that run concurrently?

Reply:
I looked at listr2 and it seems like tasks are not being processed concurrently there as well so for now I think we can keep this sequential. Parallel tasks would also need a different interface to report progress.